org.twinery.Twine.yaml

```diff
@@ -0,0 +1,69 @@
+---
+app-id: "org.twinery.Twine"
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+runtime: "org.freedesktop.Platform"
+runtime-version: "21.08"
+sdk: "org.freedesktop.Sdk"
+sdk-version: "21.08"
+command: "twine-wrapper"
+copy-icon: true
+separate-locales: false
+finish-args:
+  # Required due to being a GUI application
+  - --socket=x11
+  # Required to make sure x11 performance is achived on all platforms
+  # At least that's what the legends tell. it might be worth experimenting
+  # with dropping this permission.
+  - --share=ipc
+  # Allows to send and receive files in the Downloads directory
+  # Required until Electron supports portals for load and safe dialogs
+  - --filesystem=xdg-download
+  # Allows to safe stories (Twine uses this directory by default)
+  - --filesystem=xdg-documents/Twine:create
+cleanup:
+  - "/include"
+  - "/lib/pkgconfig"
+  - "/lib/*.la"
+  - "/lib/*.a"
+  - "/share/pkgconfig"
+  - "/share/aclocal"
+  - "/share/gtk-doc"
+  - "/share/doc"
+  - "/share/info"
+  - "/share/man"
+  - "/man"
+modules:
+  - name: twine
+    buildsystem: simple
+    build-commands:
+      - mkdir twine
+      - cd twine && unzip ../twine.zip
+      - mv twine "${FLATPAK_DEST}"
+      - install -Dm755 twine.sh /app/bin/twine-wrapper
+      - install -Dm644 org.twinery.Twine.desktop "${FLATPAK_DEST}/share/applications/org.twinery.Twine.desktop"
+      - install -Dm644 org.twinery.Twine.metainfo.xml "${FLATPAK_DEST}/share/metainfo/org.twinery.Twine.metainfo.xml"
+      - install -Dm644 org.twinery.Twine.svg "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg"
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        dest-filename: twine.zip
+        url: "https://github.com/klembot/twinejs/releases/download/2.3.15/twine_2.3.15_linux64.zip"
```

Comment:
since this is x86_64 only (which make me sad) you need a flathub.json to limit architectures.

Reply:
This should be fixed now, I'll ask upstream to provide aarch64 builds :)

Reply:
or the flatpak could be built from source....

Reply:
Nope, not gonna happen from me as long as node-based applications are not self-contained (which might happens with yarn2 at some point). Maintaining npm dependencies and debugging broken builds just to come to the same results as upstream is none of my hobbys. 🤷🏻

Reply:
I thought the whole point of Electron was to be cross platform and portable. I must be old school in what this means...

Reply:
My concern is not electron itself, but how flatpak and the node ecosystem work. From my perspective there is no point in regenerating a flatpak dependency tree from multiple package-lock.json to run the exact same commands as upstream just to have the same files around as them.

Not to mention that in either case there will be binary blobs unless I build electron as part of this flatpak, which basically includes a complete chromium build, which again takes hours to complete, as you may already know from the chromium flatpak. I don't want to put on that shoe. If that's what we we will suddenly require on flathub, I will rather withdraw this submission and build this on my GitLab and push the flatpak to a s3 storage.

Reply:
there is no requirement whatsoever.

(also for the record, I have no say in what gets accepted or reject anyway)